cppwinrt/code_writers.h

```diff
@@ -1137,7 +1137,8 @@ namespace cppwinrt
     {%
         if constexpr (!std::is_same_v<D, %>)
         {
-            auto const [_winrt_casted_result, _winrt_cast_result_code] = impl::try_as_with_reason<%, D const*>(static_cast<D const*>(this));
+            winrt::hresult _winrt_cast_result_code;
+            auto const _winrt_casted_result = impl::try_as_with_reason<%, D const*>(static_cast<D const*>(this), _winrt_cast_result_code);
```

Comment:
I'm not sure if it is still the recommended coding standard, but for a long time non-const references were disallowed and passing a pointer was preferred.  That way the call site is clearly mutating the variable because the & is visible.  In this case it is generated code so I guess it is [shrug].